docs/src/rules/no-unused-expressions.md

```diff
@@ -276,3 +276,70 @@ const myFragment = <></>;
 ```
 
 :::
+
+### TypeScript Support
+
+This rule supports TypeScript-specific expressions and follows these guidelines:
+
+1. Directives (like 'use strict') are allowed in module and namespace declarations
+2. Type-related expressions are treated as unused if their wrapped value expressions are unused:
+   * Type assertions (`x as number`, `<number>x`)
+   * Non-null assertions (`x!`)
+   * Type instantiations (`Set<number>`)
+
+Note: Although type expressions never have runtime side effects (e.g., `x!` is equivalent to `x` at runtime), they can be used to assert types for testing purposes.
+
+Examples of **correct** code for this rule when using TypeScript:
+
+::: correct
+
+```ts
+/* eslint no-unused-expressions: "error" */
+
+// Function calls are allowed
+function getSet() {
+    return Set;
+}
+
+// Type expressions wrapping function calls are allowed
```

Comment:
[Docs] The first comment confused me at first. These look like four separate examples. Maybe:

```suggestion
// Type expressions wrapping function calls are allowed
function getSet() {
    return Set;
}
```